File: aarthik_setu/lib/models/profiles/personal_profile/components/basic_details.dart

class BasicDetails {
  String salutation;
  String firstName;
  String? middleName;
  String lastName;
  DateTime dob;
  String pan;
  String gender;
  String category;
  String mobile;
  String telephone;
  String emailPersonal;
  String fatherName;
  String educationQualification;
  double netWorth;
  String nationality;
  String dependent;
  String maritalStatus;

  BasicDetails({
    required this.salutation,
    required this.firstName,
    this.middleName,
    required this.lastName,
    required this.dob,
    required this.pan,
    required this.gender,
    required this.category,
    required this.mobile,
    required this.telephone,
    required this.emailPersonal,
    required this.fatherName,
    required this.educationQualification,
    required this.netWorth,
    required this.nationality,
    required this.dependent,
    required this.maritalStatus,
  });
}

File: server/models/forms/personal_forms/basic_details.go
package personal_forms
import(
	"time"
)

type UserDetail struct {
	Salutation            string  `form:"salutation" json:"salutation"`
	FirstName            string  `form:"firstName" json:"first_name"`
	MiddleName           *string `form:"middleName" json:"middle_name,omitempty"`
	LastName             string  `form:"lastName" json:"last_name"`
	Dob                  time.Time   `form:"dob" json:"dob"`
	Pan                  string  `form:"pan" json:"pan"`
	Gender               string  `form:"gender" json:"gender"`
	Category             string  `form:"category" json:"category"`
	Mobile               string  `form:"mobile" json:"mobile"`
	Telephone            string  `form:"telephone" json:"telephone"`
	EmailPersonal        string  `form:"emailPersonal" json:"email_personal"`
	FatherName           string  `form:"fatherName" json:"father_name"`
	EducationQualification string  `form:"educationQualification" json:"education_qualification"`
	NetWorth             float64 `form:"netWorth" json:"net_worth"`
	Nationality          string  `form:"nationality" json:"nationality"`
	Dependent            string  `form:"dependent" json:"dependent"`
	MaritalStatus        string  `form:"maritalStatus" json:"marital_status"`
}
